packages/devtools_app/lib/src/charts/treemap.dart

```diff
@@ -349,77 +370,78 @@ class _NewTreemapState extends State<NewTreemap> {
   /// ```
   @override
   Widget build(BuildContext context) {
-    if (widget.rootNode == null && widget.nodes.isNotEmpty) {
+    if (rootNode == null && nodes.isNotEmpty) {
       return buildSubTreemaps();
     } else {
       return buildTreemapCell();
     }
   }
 
   Widget buildTreemapCell() {
-    if (widget.levelsVisible > 0 && widget.rootNode.children.isNotEmpty) {
+    if (levelsVisible > 0 && rootNode.children.isNotEmpty) {
       return Padding(
         padding: const EdgeInsets.all(1.0),
         child: Column(
           children: [
-            // TODO(peterdjlee): Abstract out to a widget or a method.
-            Tooltip(
-              message: widget.rootNode.name,
-              child: InkWell(
-                onTap: () {
-                  widget.onRootChangedCallback(widget.rootNode);
-                },
+            if (height > 20.0)
+              buildSelectable(
                 child: Container(
                   width: double.infinity,
-                  decoration: BoxDecoration(border: Border.all(width: 0.5)),
+                  decoration: BoxDecoration(border: Border.all()),
                   child: buildNameAndSizeText(
                     fontColor: Colors.white,
-                    onTwoLines: false,
+                    oneline: true,
                   ),
                 ),
               ),
-            ),
             Expanded(
-              child: NewTreemap.fromNodes(
-                nodes: widget.rootNode.children,
-                levelsVisible: widget.levelsVisible - 1,
-                onRootChangedCallback: widget.onRootChangedCallback,
+              child: Treemap.fromNodes(
+                nodes: rootNode.children,
+                levelsVisible: levelsVisible,
+                onRootChangedCallback: onRootChangedCallback,
+                height: height,
               ),
             ),
           ],
         ),
       );
     } else {
-      // TODO(peterdjlee): Abstract out to a widget or a method.
-      return Tooltip(
-        message: widget.rootNode.name,
-        child: InkWell(
-          onTap: () {
-            widget.onRootChangedCallback(widget.rootNode);
-          },
-          child: Container(
-            decoration: BoxDecoration(
-              color: mainUiColor,
-              border: Border.all(width: 0.5),
-            ),
-            child: Center(
-              child: buildNameAndSizeText(
-                fontColor: Colors.black,
-                onTwoLines: true,
-              ),
+      return buildSelectable(
+        child: Container(
+          decoration: BoxDecoration(
+            color: mainUiColor,
+            border: Border.all(),
+          ),
+          child: Center(
+            child: buildNameAndSizeText(
+              fontColor: Colors.black,
+              oneline: false,
             ),
           ),
         ),
       );
     }
   }
 
+  Tooltip buildSelectable({@required Widget child}) {
```

Comment:
add dart doc above this method that explains that selecting this widget will trigger a re-root of the tree to the associated rootNode